smartsim/_core/mli/comm/channel/dragon_fli.py

```diff
@@ -60,26 +55,31 @@ def __init__(
         :param sender_supplied: Flag indicating if the FLI uses sender-supplied streams
         :param buffer_size: Maximum number of sent messages that can be buffered
         """
-        super().__init__(fli_desc)
-        self._fli: "fli" = fli.FLInterface.attach(fli_desc)
-        self._channel: t.Optional["dch"] = (
-            create_local(buffer_size) if sender_supplied else None
+        descriptor = drg_util.channel_to_descriptor(fli_)
+        super().__init__(descriptor)
+
+        self._fli = fli_
+        """The underlying dragon FLInterface used by this CommChannel for communications"""
+        self._channel: t.Optional["dch.Channel"] = (
+            drg_util.create_local(buffer_size) if sender_supplied else None
         )
+        """The underlying dragon Channel used by a sender-side DragonFLIChannel
+        to attach to the main FLI channel"""
 
     def send(self, value: bytes, timeout: float = 0.001) -> None:
         """Send a message through the underlying communication channel.
 
-        :param timeout: Maximum time to wait (in seconds) for messages to send
         :param value: The value to send
+        :param timeout: Maximum time to wait (in seconds) for messages to send
         :raises SmartSimError: If sending message fails
         """
         try:
             with self._fli.sendh(timeout=None, stream_channel=self._channel) as sendh:
```

Comment:
I notice in init that self._channel can be None, is there ever a possibility that this would sendh _channel with None, if so would it just fail?

Reply:
it's possible. i have a ticket to create that will make it impossible, though. good eye.